api/v1/conditions.go

```diff
@@ -1,11 +1,12 @@
 package v1
 
 const (
-	ConditionUploaded = "Uploaded"
-	ConditionBuilt    = "Built"
-	ConditionLoaded   = "Loaded"
-	ConditionModelled = "Modelled"
-	ConditionDeployed = "Deployed"
+	ConditionReconciling = "Reconciling"
```

Comment:
Conditions here should be end-states not transition states. IMO, I would remove this and go with the suggested changes below which make this unnecessary.